yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -958,6 +957,8 @@ object Client extends Logging {
     val sparkConf = new SparkConf
 
     val args = new ClientArguments(argStrings, sparkConf)
+    // to maintan backwards-compatibility
+    sparkConf.setIfMissing("spark.executor.instances", args.numExecutors.toString)
```

Comment:
Ah, good catch. This should only be done if dynamic allocation is not enabled (to maintain compatibility for code that calls the `Client` object directly, instead of going through spark-submit).